packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -278,10 +278,7 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
           children.add(
             new Expanded(
               child: new InkResponse(
-                onTap: () {
-                  if (widget.onTap != null)
-                    widget.onTap(i);
-                },
+                onTap: widget.onTap != null ? () { widget.onTap(i); } : null,
```

Comment:
This isn't equivalent, now if you don't provide an onTap, the button won't react. That's probably fine, but we should make sure we document that.